### PR TITLE
PRIME-1525 unhook "create a signing authority" from "create an organization" flow.

### PR DESCRIPTION
**Pull request by mtpultz**
**Comments on update:**

* SA is still in form state due to dependencies with Contacts, but only updated through signing authority endpoints, and can be updated when site registration updates for Health Authorities is confirmed
* `OrganizationListViewModel` used in `SigningAuthorityController` for `GetSigningAuthorityOrganizationsByUserId`
  * Could be replaced with `SigningAuthOrgLIstViewModel`
  * Trimmed down by dropping `DisplayId`, `RegistrationId`, `HasAcceptedAgreement`, and `HasSubmittedSite`
  * `SigningAuthority` could be reduced down to `SigningAuthorityFullname` once Contacts no longer depends on it for populating the `SameAs` dropdown
  * Also could drop `Sites` and get them via a second call out to an endpoint

When dependency has been removed from Contacts:
* Create a single page services for managing get, create, and update of signing authorities to encapsulate the entire lifecycle of a signing authority into the routable page
* Drop `SigningAuthority` from view model for `SigningAuthorityFullname`

**NOTES:**

* PRIME-1131 Refactoring of Site Registration Guard was included into this ticket and points

### CODE REVIEW

**Review by AnaisHebert — APPROVED** [review]
Looks good to me, and everything still works as it should 🤸‍♀️

File: prime-dotnet-webapi/Controllers/OrganizationsController.cs [hunk 1]
```diff
@@ -44,32 +43,20 @@ public class OrganizationsController : ControllerBase
 
         // GET: api/Organizations
         /// <summary>
-        /// Gets all of the Organizations for a user, or all organizations if user has ADMIN role
+        /// Gets all of the Organizations.
         /// </summary>
         [HttpGet(Name = nameof(GetOrganizations))]
+        [Authorize(Roles = Roles.PrimeAdministrant)]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/OrganizationsController.cs:49]: +1 for remembering this, but it should be Roles.ViewSite

File: prime-dotnet-webapi/Controllers/SigningAuthorityController.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+using Prime.ViewModels;
+using Prime.ViewModels.Parties;
+
+namespace Prime.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/[controller]")]
+    [ApiController]
+    [Authorize(Roles = Roles.PrimeEnrollee + "," + Roles.ViewSite)]
+    public class SigningAuthorityController : ControllerBase
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/SigningAuthorityController.cs:21]: SigningAuthoritiesController

File: prime-dotnet-webapi/Controllers/SigningAuthorityController.cs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+using Prime.Auth;
+using Prime.Models.Api;
+using Prime.Services;
+using Prime.ViewModels;
+using Prime.ViewModels.Parties;
+
+namespace Prime.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/[controller]")]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/SigningAuthorityController.cs:18]: I think this URL should be api/parties/[controller]

File: prime-dotnet-webapi/Services/PartyService.cs [hunk 4]
```diff
@@ -66,42 +82,33 @@ public async Task<int> CreateOrUpdatePartyAsync(IPartyChangeModel changeModel, C
         public void UpdateAddress<T>(Party dbParty, T newAddress) where T : Address
         {
             var existingPartyAddress = dbParty.Addresses
-                .Where(ea => ea.Address is T)
-                .SingleOrDefault();
+                .SingleOrDefault(ea => ea.Address is T);
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/PartyService.cs:85]: 👍

**Review by james-hollinger — APPROVED** [review]
Seems L "LGTM" TM

**Pull request merged**